### ISSUE TITLE
Perpetual syncing/grabbing content

### ISSUE THREAD

**Issue by Araborum** [jangernert/FeedReader#428]
Short background:

I am running FeedReader 2.0.2 on Fedora 25. I force-updated to 2.0.2 since only a lower version is available by default for F25. Had to do it because lower version would not sync with InoReader. Now it syncs just fine, but here is the issue.

After launch, i get the reader to update the feeds, which pulls in all the new/stared articles. But the synchronisation never stops. The sync icon next to 'starred' menu item just keeps spinning and if you click it it is always stuck on (at least in my case) Grabbing full content 1/14778. It will stay like that no matter what. I don't have that many items starred or unread, even put together. It causes some minor lag and (potentially) eats at my bandwidth.

Any ideas?

**Comment by jangernert** [comment]
You should check out the log of the daemon during the sync. Just start a new instance of the 'feedreader-daemon' process in a terminal and you can see if there is some sort of crash or infinite loop or anything else that prevents the grabber to continue on.

**Comment by Araborum** [comment]
[ DEBUG ] DBusConnection: setOnline
[ DEBUG ] DBusConnection: updateSyncProgress
[ DEBUG ] DBusConnection: newFeedList
[ DEBUG ] FeedList: new FeedList
[ DEBUG ] SELECT count(*) FROM main.tags
[ DEBUG ] count 4
[ DEBUG ] initCollapseCategories
[ DEBUG ] FeedList: restore selected row: feed -76
[ DEBUG ] DBusConnection: updateSyncProgress
[ DEBUG ] articlelist-prev
[ DEBUG ] articlelist-next
[ DEBUG ] feedlist-prev
[ DEBUG ] feedlist-next
[ DEBUG ] articleview-up
[ DEBUG ] articleview-down
[ DEBUG ] articlelist-toggle-read
[ DEBUG ] articlelist-toggle-marked
[ DEBUG ] articlelist-open-url
[ DEBUG ] feedlist-mark-read
[ DEBUG ] global-sync
[ DEBUG ] articlelist-center-selected
[ DEBUG ] global-search
[ DEBUG ] global-quit
[ DEBUG ] global-help
[ DEBUG ] DBusConnection: updateArticleList
[ DEBUG ] ArticleList: updateArticleList()
[ DEBUG ] load articles from db
[ DEBUG ] SELECT ROWID, feedID, articleID, title, author, url, preview, unread, marked, tags, date, guidHash, media FROM main.articles WHERE unread = 9  ORDER BY rowid COLLATE NOCASE ASC LIMIT 12 OFFSET 0
[ DEBUG ] actual articles loaded: 0
[ DEBUG ] ArticleList.updateArticleList: newCount 0
[ DEBUG ] articlelist-prev
[ DEBUG ] articlelist-next
[ DEBUG ] feedlist-prev
[ DEBUG ] feedlist-next
[ DEBUG ] articleview-up
[ DEBUG ] articleview-down
[ DEBUG ] articlelist-toggle-read
[ DEBUG ] articlelist-toggle-marked
[ DEBUG ] articlelist-open-url
[ DEBUG ] feedlist-mark-read
[ DEBUG ] global-sync
[ DEBUG ] articlelist-center-selected
[ DEBUG ] global-search
[ DEBUG ] global-quit
[ DEBUG ] global-help
[ ERROR ] dbErrorLog: 1: no such column: makred
[ ERROR ] SELECT count(*) FROM main.articles WHERE makred = 11 AND feedID IN ("feed/http://feeds.feedburner.com/sociallending/cnry") 
[ ERROR ] no such column: makred
[ ERROR ] dbErrorLog: 21: API called with NULL prepared statement
[ ERROR ] dbErrorLog: 21: misuse at line 75611 of [29dbef4b85]
[ DEBUG ] HeaderBar: showArticleButtons true
[ DEBUG ] ContentPage.newArticleList
[ DEBUG ] ArticleList: newList
[ DEBUG ] ArticleList: disallow signals from scroll
[ DEBUG ] ArticleList: new-rows 0
[ DEBUG ] load articles from db

And then the loop starts at the SELECT ROWID line

**Comment by jangernert** [comment]
That's not the log of the feedreader-daemon as far as I can see. I spotted a typo tho :P

**Comment by Araborum** [comment]
that's a copy/paste from console of feedreader. Feedreader-daemon is below, but it seems to be dead once the below lines pass by, that's why i pasted output from feedreader initially:

[ INFO  ] Everything is logged. Used to debug!
[ INFO  ] daemon: Notification actions supported
[ ERROR ] Utils.copyAutostart: Error when getting information for file '/usr/share/FeedReader/feedreader-autostart.desktop': No such file or directory
[ INFO  ] FeedReader Daemon 2.0.2
[ DEBUG ] Opening Database: /home/XXXX/.local/share/feedreader/data/feedreader-04.db
[ ERROR ] dbErrorLog: 283: recovered 5738 frames from WAL file /home/XXXX/.local/share/feedreader/data/feedreader-04.db-wal
[ DEBUG ] database already initialized
[ DEBUG ] database already initialized
[ DEBUG ] daemon: constructor
[ DEBUG ] daemon: new FeedServer and login
[WARNING] feedserver.unloadPlugin: no plugin loaded
[ DEBUG ] feedserver: load plugin "inoreader"
[ DEBUG ] feedserver: engine load InoReader
[ DEBUG ] feedserver: plugin loaded InoReader
[ INFO  ] Inoreader: userID = 1005610801
[ DEBUG ] SELECT count(*) FROM main.CachedActions
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.CachedActions
[ DEBUG ] count 0
[ DEBUG ] CachedActionManager - executeActions: no actions to perform
[WARNING] resetCachedActions
[ DEBUG ] daemon: login status = FEED_READER_LOGIN_RESPONSE_SUCCESS
[ DEBUG ] daemon: bus aquired

**Comment by jangernert** [comment]
Of course only one instance of the daemon can run at a single time. So if you still have an instance running you need to kill that one first ;)
Also you can put your logs into tripple-"`" to render it more readable like
```
this
```

**Comment by Araborum** [comment]
Only one instance was running. I killed it, checked for any other ones and there was nothing. New output is below. the only difference it seems is that the 'recovered frames' lines number got raised by one.

[ INFO  ] Everything is logged. Used to debug!
[ INFO  ] daemon: Notification actions supported
[ ERROR ] Utils.copyAutostart: Error when getting information for file '/usr/share/FeedReader/feedreader-autostart.desktop': No such file or directory
[ INFO  ] FeedReader Daemon 2.0.2
[ DEBUG ] Opening Database: /home/dmikulenas/.local/share/feedreader/data/feedreader-04.db
[ ERROR ] dbErrorLog: 283: recovered 5739 frames from WAL file /home/dmikulenas/.local/share/feedreader/data/feedreader-04.db-wal
[ DEBUG ] database already initialized
[ DEBUG ] database already initialized
[ DEBUG ] daemon: constructor
[ DEBUG ] daemon: new FeedServer and login
[WARNING] feedserver.unloadPlugin: no plugin loaded
[ DEBUG ] feedserver: load plugin "inoreader"
[ DEBUG ] feedserver: engine load InoReader
[ DEBUG ] feedserver: plugin loaded InoReader
[ INFO  ] Inoreader: userID = 1005610801
[ DEBUG ] SELECT count(*) FROM main.CachedActions
[ DEBUG ] count 0
[ DEBUG ] SELECT count(*) FROM main.CachedActions
[ DEBUG ] count 0
[ DEBUG ] CachedActionManager - executeActions: no actions to perform
[WARNING] resetCachedActions
[ DEBUG ] daemon: login status = FEED_READER_LOGIN_RESPONSE_SUCCESS
[ DEBUG ] daemon: bus aquired

**Comment by jangernert** [comment]
Oh sorry, forgot to mention you should open up the UI once you got the daemon running inside the terminal and click the update-button to trigger a sync. Otherwise you would need to wait 10 minutes for the automatic sync (at least that's the default setting)

**Comment by sharifzadesina** [comment]
@jangernert

Sry for jumping in the middle of discuss :D

As these logs are for last version, why you do not fix this error?

`[ ERROR ] Utils.copyAutostart: Error when getting information for file '/usr/share/FeedReader/feedreader-autostart.desktop': No such file or directory`

It seems you need to just change the name of the file to `org.gnome.FeedReader-autostart.desktop` in the source code.

**Comment by Araborum** [comment]
So, after getting a monumental reel of output in the terminal, i decided to go the brute force route. i jumped in to the db directory, which in my case is /home/XXXX/.local/share/feedreader/data/

I found in there feedreader-04.*** as well as feedreader-03.*** files. I guess the later one is a remnant from the previous FeedReader version. I deleted everything apart from the feed_icons and images folders, forcing the database to be rebuilt on launch.

That seems to have fixed it very nicely: everything is working, no perpetual syncing. If i had to guess, updating to the latest version from a previous one (whichever one is available under Fedora 25) must have caused a glitch/error in the database, which led to issues like 'recovered 5738 frames from WAL file ' and the never ending syncing problem.

So there you have it, problem solved!
